lib/utilities/collectCorsHeaders.js

```diff
@@ -0,0 +1,34 @@
+import { findCorsRule,
+generateCorsResHeaders } from '../api/apiUtils/object/corsResponse.js';
+
+/**
+ * collectCorsHeaders - gather any relevant CORS headers
+ * @param {object} request - http request object
+ * @param {BucketInfo} bucket - instance of BucketInfo class
+ * @return {object} - object containing CORS headers
+ */
+export default function collectCorsHeaders(request, bucket) {
```

Comment:
I would suggest sending `httpOrigin`, `httpMethod` as params to this method instead of the entire request object. This would make it easy on us in the future when we remove the references to request object.

Edit: Sorry, I know that this would be a pain to update its references everywhere but it's a good coding pattern.